src/Common/tests/System/PlatformDetection.cs

```diff
@@ -23,6 +23,32 @@ public static class PlatformDetection
 
         public static int WindowsVersion { get; } = GetWindowsVersion();
 
+        private static Lazy<bool> m_isWindowsSubsystemForLinux = new Lazy<bool>(GetIsWindowsSubsystemForLinux);
+
+        public static bool IsWindowsSubsystemForLinux => m_isWindowsSubsystemForLinux.Value;
+        public static bool IsNotWindowsSubsystemForLinux => !IsWindowsSubsystemForLinux;
+
+        private static bool GetIsWindowsSubsystemForLinux()
+        {
+            // https://github.com/Microsoft/BashOnWindows/issues/423#issuecomment-221627364
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                const string versionFile = "/proc/version";
+                if (File.Exists(versionFile))
+                {
+                    var s = File.ReadAllText(versionFile);
+
+                    if (s.Contains("Microsoft") || s.Contains("WSL"))
```

Comment:
I'm not sure...WSL sounds like an uncommon sequence of characters to me, but that's hard to predict.  We could maybe do a little more rigorous parsing here, but I'm not sure it's worth the additional risk of getting things wrong.  The worst-case scenario is that we stop running some tests on a distro where they *should* run, but at least there's a chance that whoever is porting to that distro will notice, since xunit prints messages about tests that were skipped conditionally.